rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSink.java

```diff
@@ -87,7 +94,8 @@ public void open(Configuration parameters) throws Exception {
         Validate.notNull(serializationSchema, "KeyValueSerializationSchema can not be null");
 
         producer = new DefaultMQProducer();
-        producer.setInstanceName(String.valueOf(getRuntimeContext().getIndexOfThisSubtask()));
+
+        producer.setInstanceName(this.instanceName.orElse(String.valueOf(getRuntimeContext().getIndexOfThisSubtask())));
```

Comment:
There are usually more than one threads/tasks in one process in the distributed environment. So you cannot use same instanceName for an operator. 
The following is a candidate:

> producer.setInstanceName(this.instanceName + String.valueOf(getRuntimeContext().getIndexOfThisSubtask()));

Reply:
Thanks. It's a simpler way to resolve the problem: create clients for each sink and source.
The PR has been updated.